odoo/addons/base/models/ir_topbar_actions.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+from ast import literal_eval
+
+class IrActionsTopbar(models.Model):
+    _name = 'ir.actions.topbar'
+    _description = 'TopBar Actions'
+    _table = 'ir_act_topbar'
+    _order = 'sequence'
+
+    parent_action_id = fields.Many2one('ir.actions.act_window', string='Parent Action', ondelete="cascade")
+    name = fields.Char(string='Action Name', required=False, translate=True)
+    sequence = fields.Integer()
+    res_id = fields.Integer(string="Active Parent Id")
+    res_model = fields.Char(string='Active Parent Model', required=True)
+    # It is required to have either action_id or python_action
+    action_id = fields.Many2one('ir.actions.act_window', string="Action Id", ondelete="cascade")
+    python_action = fields.Char(string="Python Action")
+    user_id = fields.Many2one('res.users', string="User specific topbar action. If empty, shared topbar action", ondelete="cascade")
+    is_deletable = fields.Boolean(compute="_compute_is_deletable")
+    default_view_mode = fields.Char(string="Default view (if none, default view of the action is taken)")
+    filter_ids = fields.One2many("ir.filters", "topbar_action_id", string="Default filter of the topbar action (if none, no filters)")
+    is_visible = fields.Boolean(string="Computed field to check if the record should be visible acording to the domain", compute="_compute_is_visible")
+    domain = fields.Char(string='Domain Value', default="[]",
+                         help="Domain applied to the current Id of the Parent Model")
+    context = fields.Char(string='Context Value', default={}, required=True,
+                          help="Context dictionary as Python expression, empty by default (Default: {})")
+
+    @api.constrains('action_id', 'python_action')
+    def _check_only_one_action_defined(self):
+        for record in self:
+            if bool(record.action_id) == bool(record.python_action):
+                raise ValidationError("You cannot define a xml action and a python action for the same topbar action.")
+            elif record.python_action and not record.name:
+                raise ValidationError("You cannot define a xml python action and not define the name.")
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if "action_id" in vals and isinstance(vals["action_id"], list):
+                vals["action_id"] = vals["action_id"][0]
+            if "parent_action_id" in vals and isinstance(vals["parent_action_id"], list):
+                vals["parent_action_id"] = vals["parent_action_id"][0]
```

Comment:
What is that for?

Reply:
It is because the values for the action_id or parent_action_id sent by the client are structrured as [id, name]. So we need to take only the id to create the new topbar action without error

Reply:
I guess the call is in your code. Can't we change it?
What is usually the value of a many2one in the vals of a create?

Reply:
Ok, so in order to not do that, I changed something in the control_panel.js. (to be edited with the new lines)